### PR TITLE
bclconvert: Add version arg and move download to Dockerfile

### PR DESCRIPTION
**Pull request by edmundmiller**
https://github.com/nf-core/modules/pull/4823/commits/8249312f334a9f60750d4a654db0d707113ccf70 explains most of it

Less manual steps the better

Inspired by #4818

### CODE REVIEW

**Review by maxulysse — APPROVED** [review]
Love it, we should have done that earlier

**Review by matthdsm — APPROVED** [review]
Looking good! 👍

**Comment by edmundmiller** [comment]
Pushed this version over the other one just so we're not out of sync. Let's see if it breaks anything 😬 
image

File: modules/nf-core/bclconvert/Dockerfile [hunk 1]
```diff
@@ -2,16 +2,29 @@
...
+    && rm -rf /var/lib/apt/lists/*
 # Link hostname cmd to fix hardcoded path
 RUN ln -s /bin/hostname /usr/bin/hostname
-COPY bcl-convert /usr/local/bin/bcl-convert
-RUN chmod +x /usr/local/bin/bcl-convert
+# Install bcl-convert
+ADD https://s3.amazonaws.com/webdata.illumina.com/downloads/software/bcl-convert/bcl-convert-${BCLCONVERT_VERSION}-2.el7.x86_64.rpm bcl-convert.rpm
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN rpm2cpio bcl-convert.rpm | cpio -i --make-directories \
+    && rm bcl-convert.rpm
```
Comment (pontus) [inline modules/nf-core/bclconvert/Dockerfile:30]:
```suggestion
RUN rpm2cpio bcl-convert.rpm | cpio -i --make-directories \
    && rm bcl-convert.rpm && rm -rf /var/log
```

Comment (edmundmiller) [reply modules/nf-core/bclconvert/Dockerfile]: Burn it all down! 🔥

**Pull request merged**